### ISSUE TITLE
Installation error when installing on a path with spaces using Bundler

### ISSUE THREAD

**Issue by bomberstudios** [tmm1/http_parser.rb#47]
Steps:
- create a folder with spaces on its path (i.e: `~/http parser/`)
- add `http_parser.rb` to a `Gemfile`:

```
source 'https://rubygems.org'

gem 'http_parser.rb'
```
- run `bundle install --path _vendor`

What happens:

```
Fetching gem metadata from https://rubygems.org/.
Fetching version metadata from https://rubygems.org/
Resolving dependencies...
Installing http_parser.rb 0.6.0 with native extensions
Using bundler 1.13.2
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: /Users/ale/Desktop/tmp/http parser/_vendor/gems/http_parser.rb-0.6.0/ext/ruby_http_parser
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby -r ./siteconf20161005-24321-171301n.rb extconf.rb
creating Makefile

current directory: /Users/ale/Desktop/tmp/http parser/_vendor/gems/http_parser.rb-0.6.0/ext/ruby_http_parser
make "DESTDIR=" clean

current directory: /Users/ale/Desktop/tmp/http parser/_vendor/gems/http_parser.rb-0.6.0/ext/ruby_http_parser
make "DESTDIR="
compiling ruby_http_parser.c
clang: error: no such file or directory: 'parser/_vendor/gems/http_parser.rb-0.6.0/ext/ruby_http_parser'
make: *** [ruby_http_parser.o] Error 1

make failed, exit code 2

Gem files will remain installed in /Users/ale/Desktop/tmp/http parser/_vendor/gems/http_parser.rb-0.6.0 for
inspection.
Results logged to /Users/ale/Desktop/tmp/http
parser/_vendor/extensions/universal-darwin-16/2.0.0/http_parser.rb-0.6.0/gem_make.out

An error occurred while installing http_parser.rb (0.6.0), and Bundler cannot continue.
Make sure that `gem install http_parser.rb -v '0.6.0'` succeeds before bundling.
```

Renaming the folder to `http_parser` fixes the issue.

**Comment by bomberstudios** [comment]
(not sure if the project is even alive, but I thought I'd file)

**Comment by jaysoifer** [comment]
I was going through this and changing the folder name fixed it.

How did you find out it was a directory name issue?

**Comment by bomberstudios** [comment]
> How did you find out it was a directory name issue?

It's not the first time I get errors with folder names in Ruby (and also, the `error: no such file or directory: 'parser/_vendor/gems/http_parser.rb-0.6.0/ext/ruby_http_parser'` bit on the logs was a bit suspicious : )

**Comment by MarisolenlaWeb** [comment]
This is still happening, I lost 3 hours today because of this bug. Please fix.

**Comment by AntonioMateoGomez** [comment]
I had the same problem and I solved it by removing a space in the name of a folder from my personal path

**Comment by dobrowins** [comment]
> I had the same problem and I solved it by removing a space in the name of a folder from my personal path

That helped. It is such a disgrace though.

**Comment by silambarasanmkce** [comment]
> I had the same problem and I solved it by removing a space in the name of a folder from my personal path
 `require': incompatible library version - /Users/silambarasan/granville-station-web/.bundle/ruby/2.6.0/gems/http_parser.rb-0.6.0/lib/ruby_http_parser.bundle (LoadError)


how can i fix the error

**Comment by tsubasakat** [comment]
@silambarasanmkce

If your project’s parent directory name contains spaces, this error will occur.

Please change the name of the directory and try again.

**Comment by Overload119** [comment]
Another workaround:

https://rubygems.org/gems/http_parser.rb is only up to date on 0.6.0
the latest version is 0.6.1 which fixes this issue.

Adding this line to `Gemfile` resolves the issue for me.
`gem 'http_parser.rb', git: "git@github.com:tmm1/http_parser.rb.git"`

**Comment by tmm1** [comment]
Is anyone interested in taking over maintenance if this project and rubygem releases?

**Comment by helenatxu** [comment]
> I had the same problem and I solved it by removing a space in the name of a folder from my personal path

Same here, ruby 2.3.4, rails 5.2, Catalina Mac OS.

**Comment by ImaCrea** [comment]
same problem here. lost many hours on that!

**Comment by drewbroadley** [comment]
Hi all, Mac OSX user here where my folders were syncing with iCloud (and had a hard coded space in folder I couldn't change). My solution:

Add to Gemfile:
`gem 'http_parser.rb', git: "https://github.com/tmm1/http_parser.rb.git"`

The one above using ssh:// didn't work, this one did.

**Comment by echobash** [comment]
I faced the same issue.
If `pwd` has anywhere space in the path,the same error is thrown.
Once I changed the name of on of the directory after removing the space,it worked like charm.

**Comment by sts-ryan-holton** [comment]
I'm getting the following error when trying to run `gem install jekyll`, not sure how to fix or what to change if related:

```
Building native extensions. This could take a while...
ERROR:  Error installing jekyll:
    ERROR: Failed to build gem native extension.

    current directory: /usr/local/rvm/gems/ruby-2.7.0/gems/http_parser.rb-0.6.0/ext/ruby_http_parser
/usr/local/rvm/rubies/ruby-2.7.0/bin/ruby -I /usr/local/rvm/rubies/ruby-2.7.0/lib/ruby/2.7.0 -r ./siteconf20200811-15255-1rx4cpi.rb extconf.rb
creating Makefile

current directory: /usr/local/rvm/gems/ruby-2.7.0/gems/http_parser.rb-0.6.0/ext/ruby_http_parser
make "DESTDIR=" clean

current directory: /usr/local/rvm/gems/ruby-2.7.0/gems/http_parser.rb-0.6.0/ext/ruby_http_parser
make "DESTDIR="
compiling ruby_http_parser.c
compiling ryah_http_parser.c
linking shared-object ruby_http_parser.so
/usr/bin/ld: unrecognized option '--compress-debug-sections=zlib'
/usr/bin/ld: use the --help option for usage information
collect2: error: ld returned 1 exit status
make: *** [ruby_http_parser.so] Error 1

make failed, exit code 2

Gem files will remain installed in /usr/local/rvm/gems/ruby-2.7.0/gems/http_parser.rb-0.6.0 for inspection.
Results logged to /usr/local/rvm/gems/ruby-2.7.0/extensions/x86_64-linux/2.7.0/http_parser.rb-0.6.0/gem_make.out
```

I don't think any of the above fixes it for me.

**Comment by p2635** [comment]
When I try to run bundle install on a jekyll website project, I get the same thing. I noticed that it's installing to '.gem ' (note the space) in my home dir instead of simply 'gem'. I guess this is the issue, can someone help me?

```
phil@computer: bundle install                  
Fetching gem metadata from https://rubygems.org/..........
Fetching gem metadata from https://rubygems.org/.
Resolving dependencies...
Using public_suffix 4.0.6
Using addressable 2.7.0
Using bundler 2.1.4
Using colorator 1.1.0
Using concurrent-ruby 1.1.7
Using eventmachine 1.2.7
Fetching http_parser.rb 0.6.0
Installing http_parser.rb 0.6.0 with native extensions
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: /home/phil/.gem /ruby/2.7.0/gems/http_parser.rb-0.6.0/ext/ruby_http_parser
/usr/bin/ruby2.7 -I /usr/lib/ruby/2.7.0 -r ./siteconf20201128-3772259-kdqmdm.rb extconf.rb
creating Makefile

current directory: /home/phil/.gem /ruby/2.7.0/gems/http_parser.rb-0.6.0/ext/ruby_http_parser
make "DESTDIR=" clean

current directory: /home/phil/.gem /ruby/2.7.0/gems/http_parser.rb-0.6.0/ext/ruby_http_parser
make "DESTDIR="
compiling ruby_http_parser.c
gcc: error: /ruby/2.7.0/gems/http_parser.rb-0.6.0/ext/ruby_http_parser: No such file or directory
make: *** [Makefile:245: ruby_http_parser.o] Error 1

make failed, exit code 2

Gem files will remain installed in /home/phil/.gem /ruby/2.7.0/gems/http_parser.rb-0.6.0 for inspection.
Results logged to /home/phil/.gem /ruby/2.7.0/extensions/x86_64-linux/2.7.0/http_parser.rb-0.6.0/gem_make.out

An error occurred while installing http_parser.rb (0.6.0), and Bundler cannot continue.
Make sure that `gem install http_parser.rb -v '0.6.0' --source 'https://rubygems.org/'` succeeds before bundling.

In Gemfile:
  minima was resolved to 2.5.1, which depends on
    jekyll-feed was resolved to 0.15.1, which depends on
      jekyll was resolved to 4.1.1, which depends on
        em-websocket was resolved to 0.5.2, which depends on
          http_parser.rb
```

So I tried to do what it told me to do and ran gem install.

```
phil@computer: gem install http_parser.rb
Building native extensions. This could take a while...
Successfully installed http_parser.rb-0.6.0
Parsing documentation for http_parser.rb-0.6.0
unknown encoding name "chunked\r\n\r\n25" for ext/ruby_http_parser/vendor/http-parser-java/tools/parse_tests.rb, skipping
Done installing documentation for http_parser.rb after 0 seconds
1 gem installed
```

What is this encoding issue? I don't understand how I can fix it.

If I run `bundle install` again at this point then it does the same thing as above.

**Comment by ashie** [comment]
I've released v0.6.1.
v0.6.1 will fix this issue.
